packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -128,20 +152,52 @@ export function getChainIdsCaveat(
     return null;
   }
 
-  assert(permission.caveats.length === 1);
-  assert(permission.caveats[0].type === SnapCaveatType.ChainIds);
+  const caveat = permission.caveats.find(
+    (permCaveat) => permCaveat.type === SnapCaveatType.ChainIds,
+  );
+
+  assert(caveat);
```

Comment:
Do we need to assert here or can we just return null if it is not found?

Right now we do both, which seems unnecessary. 

Same applies to `getLookupMatchersCaveat`

Reply:
Seems simpler to just return null if the caveat is optional anyway?

Reply:
true